internal/filegreeter/greeter.go

```diff
@@ -0,0 +1,63 @@
+package filegreeter
+
+import (
+	"bufio"
+	"context"
+	"github.com/Tommy647/go_example/internal/greeter"
+	"io"
+	"log"
+	"os"
+)
+
+// New returns a new instance of our file greeter
+func New(path string) *FileGreeter {
+	file, err := os.Open(path)
+	defer file.Close()
+	if err != nil {
+		panic("Error opening file" + err.Error())
```

Comment:
and Line 44

Don't use `log.Fatal` or `panic` in a 'library' or internal package, they will force the application to shutdown, not just the request they were call in

Return an error and let the calling code decided how to handle it

Reply:
Makes sense, have removed these and passed the error up